label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -113,6 +115,25 @@ public void getLabelApplicationRules(
     }
   }
 
+  @Override
+  public void getLabelApplicationRule(
+      GetLabelApplicationRuleRequest request,
+      StreamObserver<GetLabelApplicationRuleResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      LabelApplicationRule rule =
+          this.labelApplicationRuleStore
+              .getData(requestContext, request.getId())
```

Comment:
This would also need to respect the deleted system IDs (another benefit from sharing the existing RPC)